compiler/damlc/daml-preprocessor/src/DA/Daml/Preprocessor.hs

```diff
@@ -161,6 +161,14 @@ checkRecordConstructor (GHC.L _ m) = mapMaybe getRecordError (GHC.hsmodDecls m)
         | otherwise
         = Nothing
 
+    isNewType :: GHC.NewOrData -> Bool
+    isNewType = (GHC.NewType ==)
+
+    isRecCon :: GHC.HsConDeclDetails GHC.GhcPs -> Bool
+    isRecCon = \case
+        GHC.RecCon{} -> True
+        _ -> False
+
     message tyNameStr conNameStr = unwords
         [ "Record type", tyNameStr, "has constructor", conNameStr
```

Comment:
How about using `newtype` in the error message (only where it applies of course)? It might not be obvious to everyone that a newtype is a record.

Reply:
Good idea. :)